### PR TITLE
add usage to kubeclient manager

### PR DESCRIPTION
**Pull request by pdgetrf**
### Changes
Started to migrated usages to kubeclient manager. This is the 2nd part of the kubelet changes for scale-out. The next part will complete all the changes.

### Note:
Tests in this PR (and next) are mostly for backward compatible. Will add more scale-out specific tests after all POC changes are in.

### Validation
1. added and fixed unit tests
2. Arktos-up came up correctly
3. Created and deleted the follow resources correctly
```
kubectl create tenant zeth
kubectl create -f /tmp/web.yaml --tenant zeth
kubectl create deployment nginx --image=nginx --tenant zeth
kubectl create -f vanilla.yaml --tenant zeth

kubectl create tenant apple
kubectl create -f /tmp/web.yaml --tenant apple
kubectl create deployment nginx --image=nginx --tenant apple
kubectl create -f vanilla.yaml --tenant apple
```
4. 100-node kube-mark cluster up correctly

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -312,8 +314,17 @@ func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, ku
...
+			source := kubetypes.ApiserverSource + strconv.Itoa(i)
+			klog.Infof("Watching apiserver %s", source)
+			uc := cfg.Channel(source)
+			config.NewSourceApiserver(client, nodeName, uc)
+		}
+	}
+
 	if kubeDeps.KubeClient != nil {
-		klog.Infof("Watching apiserver")
+		klog.Info("Watching apiserver")
```
Comment (chenqianfzh) [inline pkg/kubelet/kubelet.go:327]: Just curious, why change Infof to Info?

Comment (pdgetrf) [reply pkg/kubelet/kubelet.go:327]: func Infof(format string, args ...interface{})
this my understanding is to take a format with parameters whereas Info does not.

File: pkg/kubelet/kubelet_test.go [hunk 2]
```diff
@@ -236,12 +241,20 @@ func newTestKubeletWithImageList(
...
 	kubelet.podManager = kubepod.NewBasicPodManager(fakeMirrorClient, kubelet.secretManager, kubelet.configMapManager, podtest.NewMockCheckpointManager())
-	kubelet.statusManager = status.NewManager(fakeKubeClient, kubelet.podManager, &statustest.FakePodDeletionSafetyProvider{})
+	kubelet.statusManager = status.NewManager(kubelet.kubeTPClients, kubelet.podManager, &statustest.FakePodDeletionSafetyProvider{})
+
+	kubeclientmanager.NewKubeClientManager()
+	kubeclientmanager.ClientManager.RegisterTenantSourceServer(
+		"api",
+		&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Tenant: "system",
```
Comment (chenqianfzh) [inline pkg/kubelet/kubelet_test.go:256]: "system" -> metav1.TenantSystem

File: pkg/kubelet/kubelet_test.go [hunk 3]
```diff
@@ -1993,6 +2009,14 @@ func TestHandlePodResourcesResize(t *testing.T) {
 	defer kubelet.podManager.DeletePod(testPod2)
 	defer kubelet.podManager.DeletePod(testPod1)
 
+	kubeclientmanager.NewKubeClientManager()
+	kubeclientmanager.ClientManager.RegisterTenantSourceServer(
+		"api",
+		&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Tenant: "system",
+			}})
```
Comment (chenqianfzh) [inline pkg/kubelet/kubelet_test.go:2018]: same comment as above.

Besides, do we need non-system tenant test cases?

File: pkg/kubelet/pod/mirror_client_test.go [hunk 4]
```diff
@@ -54,3 +57,76 @@ func TestParsePodFullName(t *testing.T) {
...
+				},
+			},
+			expect: true,
+		},
+	}
+
+	for i, test := range testcases {
+		actual := IsStaticPod(test.pod)
+		if test.expect != actual {
+			t.Errorf("case %d faile: expected %v, got %v", i, test.expect, actual)
```
Comment (chenqianfzh) [inline pkg/kubelet/pod/mirror_client_test.go:129]: typo "faile"

File: pkg/kubelet/pod/mirror_client_test.go [hunk 5]
```diff
@@ -54,3 +57,76 @@ func TestParsePodFullName(t *testing.T) {
 		}
 	}
 }
+
+func TestIsStaticPod(t *testing.T) {
+	testcases := []struct {
+		pod    *v1.Pod
+		expect bool
```
Comment (chenqianfzh) [inline pkg/kubelet/pod/mirror_client_test.go:64]: rename "expect" -> "expectStaticPod"?

File: pkg/kubelet/kubelet_test.go [hunk 6]
```diff
@@ -1993,6 +2009,14 @@ func TestHandlePodResourcesResize(t *testing.T) {
 	defer kubelet.podManager.DeletePod(testPod2)
 	defer kubelet.podManager.DeletePod(testPod1)
 
+	kubeclientmanager.NewKubeClientManager()
+	kubeclientmanager.ClientManager.RegisterTenantSourceServer(
+		"api",
```
Comment (chenqianfzh) [inline pkg/kubelet/kubelet_test.go:2014]: Let's use the constant instead of hard-coded "api".

File: pkg/kubelet/types/pod_update_test.go [hunk 7]
```diff
@@ -218,3 +218,76 @@ func TestIsCriticalPodBasedOnPriority(t *testing.T) {
 		}
 	}
 }
+
+func TestIsStaticPod(t *testing.T) {
+	testcases := []struct {
+		pod    *v1.Pod
+		expect bool
```
Comment (chenqianfzh) [inline pkg/kubelet/types/pod_update_test.go:225]: ”expect“ -> "staticPod"

**Review by chenqianfzh — APPROVED** [review]
Look good to me overall. I only have minor comments, such as variable naming.

File: pkg/kubelet/kubelet_test.go (hunk 2 above)
Comment (pdgetrf) [reply pkg/kubelet/kubelet_test.go:256]: fixed

File: pkg/kubelet/pod/mirror_client_test.go (hunk 4 above)
Comment (pdgetrf) [reply pkg/kubelet/pod/mirror_client_test.go:129]: sin of copy paste. fixed here an 3 other places.

File: pkg/kubelet/pod/mirror_client_test.go (hunk 5 above)
Comment (pdgetrf) [reply pkg/kubelet/pod/mirror_client_test.go:64]: debatable. I feel together with the 'actual' and function name, "expect" in this context should be self-explanatory in meaning.

File: pkg/kubelet/types/pod_update_test.go (hunk 7 above)
Comment (pdgetrf) [reply pkg/kubelet/types/pod_update_test.go:225]: same argument as the comment above.

File: pkg/kubelet/kubelet_test.go (hunk 6 above)
Comment (pdgetrf) [reply pkg/kubelet/kubelet_test.go:2014]: fixed

File: pkg/kubelet/kubelet_test.go (hunk 3 above)
Comment (pdgetrf) [reply pkg/kubelet/kubelet_test.go:2018]: kubelet_test does not have any tenant test yet and it should need them. I'll create an issue to track this.

**Pull request merged**